cmd/tailscaled/tailscaled.go

```diff
@@ -141,6 +206,10 @@ func main() {
 		log.Fatalf("tailscaled does not take non-flag arguments: %q", flag.Args())
 	}
 
+	if args.statepath == paths.DefaultTailscaledStateFile() {
+		tryWindowsAppDataMigration()
```

Comment:
I think this call is at the wrong stage of the Windows service lifecycle. See the "if beWindowsSubprocess" up on line 200. That's the path where we capture logs.

So you have the wrong process doing this migration.

This way is only taken when running tailscaled.exe by hand in a cmd.exe window or using the tswin debug tool.

Reply:
Should tryWindowsAppDataMigration() be called from beWindowsSubprocess() right before it calls startIPNServer() ?

Reply:
Ideally we'd place it in one place that's shared by both code paths: the foreground tailscaled mode on Windows as well as the service mode.

ipnserver.Run perhaps?

I think I'd make DefaultTailscaledStateFile not do any fallback work; just have it return the new path only. 

The fallback work could probably be in ipnsever.Run before it calls ipn.NewFileStore

Reply:
Moved to ipnserver.Run, moved the fallback into ipnserver.
Verified that the log message appears now: `2021-09-17 16:11:58.8096887 -0700 PDT: 2021/09/17 16:11:58 tryWindowsAppDataMigration: successfully migrated: from C:\Windows\system32\config\systemprofile\AppData\Local\Tailscale\server-state.conf to C:\ProgramData\Tailscale\server-state.conf`